src/tools/genpolicy/src/registry_containerd.rs

```diff
@@ -60,17 +61,42 @@ impl Container {
             .await
             .unwrap();
         let image_layers = get_image_layers(
-            layers_cache_file_path,
+            config.layers_cache_file_path.clone(),
             &manifest,
             &config_layer,
             &ctrd_client,
         )
         .await?;
 
+        // Find the last layer with an /etc/* file, respecting whiteouts.
+        let mut passwd = String::new();
+        let mut group = String::new();
+        // Nydus/guest_pull doesn't make available passwd/group files from layers properly.
+        // See issue https://github.com/kata-containers/kata-containers/issues/11162
+        if !config.settings.cluster_config.guest_pull {
+            for layer in &image_layers {
+                if layer.passwd == WHITEOUT_MARKER {
+                    passwd = String::new();
+                } else if !layer.passwd.is_empty() {
+                    passwd = layer.passwd.clone();
+                }
+
+                if layer.group == WHITEOUT_MARKER {
+                    group = String::new();
+                } else if !layer.group.is_empty() {
+                    group = layer.group.clone();
+                }
+            }
+        } else {
+            warn!("Guest pull is enabled, skipping passwd/group file parsing");
```

Comment:
Imho the `warn` level is too high - this is not a malfunction we're working around, but the consequence of a config setting.

Reply:
Made it `info`